src/components/escrowActions.jsx

```diff
@@ -0,0 +1,105 @@
+import React, { useEffect, useState } from 'react';
+import { Button, Flex, Spinner, Text } from '@chakra-ui/react';
+
+import { useUser } from '../contexts/UserContext';
+import { MinionService } from '../services/minionService';
+import { useOverlay } from '../contexts/OverlayContext';
+import { createPoll } from '../services/pollService';
+import { useTX } from '../contexts/TXContext';
+
+const EscrowActions = ({ proposal, address, injectedProvider, daochain }) => {
+  const { errorToast, successToast, setTxInfoModal } = useOverlay();
+  const { molochAddress, minionAddress, proposalId } = proposal;
+  const { cachePoll, resolvePoll } = useUser();
+  const [loading, setLoading] = useState(true);
+  const [tokensAvailable, setTokensAvailable] = useState(false);
+  const { refreshDao } = useTX();
+
+  useEffect(() => {
+    if (loading) {
+      MinionService({
+        web3: injectedProvider,
+        minion: minionAddress,
+        chainID: daochain,
+        minionType: 'escrowMinion',
+      })('escrowBalances')({
+        args: [molochAddress, proposalId, 0],
+      })
+        .then(({ executed }) => {
+          setTokensAvailable(!executed);
+          setLoading(false);
+        })
+        .catch(err => {
+          console.error(err);
+          setLoading(false);
+        });
+    }
+  }, []);
+
+  const withdrawEscrowTokens = async () => {
+    const args = [proposalId, molochAddress, [0]];
+
+    try {
+      setLoading(true);
+      const poll = createPoll({ action: 'withdrawEscrowTokens', cachePoll })({
```

Comment:
I think we want to change this to use the new transaction legos.

here is an example doing a withdraw on the moloch contract:
https://github.com/HausDAO/daohaus-app/blob/develop/src/components/withdraw.jsx#L18

still make a tx lego - and load that directly into the submitTransaction function